### DISCUSSION TITLE
LessonsNotes/D2CX_AdvancedDataStructure/Chap02/

### DISCUSSION THREAD

**Discussion by giscus** [sqc-cyh/MyWeb-discussion#9] [General]
# LessonsNotes/D2CX_AdvancedDataStructure/Chap02/

https://sh17c.top/LessonsNotes/D2CX_AdvancedDataStructure/Chap02/

**Comment by antelo111** [comment]
红黑树插入操作中case1换完颜色不继续像case3那样往下做的原因应该是黑高吧，因为case1和case3的初始情况不一样；
case1就是把问题不断往上传递，case3是能把问题最终解决的

**Reply by sqc-cyh** [reply]
你说的是正确滴，我笔记中的意思是插入时通过case1可能会变换出case2或者case3的情况，但需要“向上”去看，而不是把节点向下去换，从我引用的修佬的状态机的图中可以看到，case3是能够把问题最终解决的，感谢您的补充~
